Security (#7)
tjav/upskillingquests: Security/Basics/security-basics-5.md

# Security-Basics-5 Governance with Policies

## Description

> One of the [design principles](https://docs.microsoft.com/en-us/azure/cloud-adoption-framework/ready/enterprise-scale/design-principles) of the [Enterprise Scale Landing Zones](https://docs.microsoft.com/en-us/azure/cloud-adoption-framework/ready/enterprise-scale/architecture#:~:text=Enterprise-scale%20is%20an%20architectural%20approach%20and%20a%20reference,roadmap%20and%20the%20Cloud%20Adoption%20Framework%20for%20Azure.) is the Policy driven governance. The idea is to enable democratization of subscriptions and therefore application owner DevOps teams to be able to choose thier way of work, tooling etc. by enforcing guardrails instead of restricting them on their choices of tooling or describing how and what they can do. [Azure Policy](https://docs.microsoft.com/en-us/azure/governance/policy/overview) helps with definign such guard rails.

Reviewer: ```suggestion
One of the [design principles](https://docs.microsoft.com/en-us/azure/cloud-adoption-framework/ready/enterprise-scale/design-principles) of the [Enterprise Scale Landing Zones](https://docs.microsoft.com/en-us/azure/cloud-adoption-framework/ready/enterprise-scale/architecture#:~:text=Enterprise-scale%20is%20an%20architectural%20approach%20and%20a%20reference,roadmap%20and%20the%20Cloud%20Adoption%20Framework%20for%20Azure.) is the Policy driven governance. The idea is to enable democratization of subscriptions and therefore application owner DevOps teams to be able to choose their way of work, tooling etc. by enforcing guardrails instead of restricting them on their choices of tooling or describing how and what they can do. [Azure Policy](https://docs.microsoft.com/en-us/azure/governance/policy/overview) helps with defining such guardrails.
```

> THe purpose of this quest is to explore the wcapabilities of Azure Policies.

Reviewer: ```suggestion
The purpose of this quest is to explore the capabilities of Azure Policies.
```

## Outcome

> Use the built-in 'Secure transfer to storage accounts should be enabled' Azure Policy to achieve the followings:

Reviewer: ```suggestion
Use the built-in 'Secure transfer to storage accounts should be enabled' Azure Policy to achieve the following:
```

- Assign the Policy to your subscription or to a resource group in your environment in Audit mode
- Create a Storage Account. After deployment go to the 'Configuration' point under 'Settings' for the Storage account and select 'Disabled' for 'Secure transfer required'.
- Go back to Azure Policies in the Portal and select the Compliance section. Wait until the non-compliant resources show up, you might need to refresh and wait up to an hour or so. Then change the setting back to 'Enabled'.
- Change the Policy assignment to 'Deny'
- Go back to the storage account and go to the 'Configuration' point under 'Settings' for the Storage account and select 'Disabled' for 'Secure transfer required'.

## Review and discussion
When would you use Audit? When would you use Deny? 
What is the difference between a Policy and an Initiative?

> When would you assing a Policy? When would you assing an Initiative?

Reviewer: ```suggestion
When would you assign a Policy? When would you assign an Initiative?
```

If you want to talk through or validate your quest after you completed, contact [Matyas Safranka](mailto:matyas@microsoft.com) via email or Teams.

## References

> It is not mandantory to use these references.

Reviewer: ```suggestion
It is not mandatory to use these references.
```
